src/sql/src/connectors.rs

```diff
@@ -0,0 +1,218 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use mz_repr::GlobalId;
+use mz_sql_parser::ast::{
+    AstInfo, AvroSchema, CreateSourceConnector, CreateSourceFormat, CreateSourceStatement,
+    CsrConnector, CsrConnectorAvro, CsrConnectorProto, Format, KafkaConnector,
+    KafkaSourceConnector, ProtobufSchema, UnresolvedObjectName,
+};
+
+use crate::normalize::unresolve;
+use crate::normalize::unresolved_object_name;
+use crate::plan::PlanError;
+use crate::plan::StatementContext;
+use crate::{catalog::SessionCatalog, normalize};
+
+/// Uses the provided catalog to populate all Connector references with the values of the connector
+/// it references allowing it to be used as if there was no indirection
+pub fn reify_connectors<T: AstInfo>(
```

Comment:
I agonized over naming this but felt that `resolve` was already heavily used and wanted to avoid confusion and thought that this does actually specify pretty clearly what's happening by taking an abstract reference and transforming it to concrete values.

That said I am very loosely attached to this specific term and will gladly adopt something that will be more effective. Perhaps `populate` or `hydrate`?